code/modules/projectiles/gun_attachables.dm

```diff
@@ -1756,3 +1767,37 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	for(var/X in master_gun.actions)
 		var/datum/action/A = X
 		A.update_button_icon()
+
+//Initially made for
+/obj/item/attachable/barrel_mod
+	name = "A barrel modifier"
+	desc = "A modification for your gun barrel"
+	icon_state = ""
+	attach_icon = ""
+	slot = ATTACHMENT_BARREL_MOD
+	flags_attach_features = ATTACH_REMOVABLE
+	pixel_shift_x = 0
+	pixel_shift_y = 0
+	size_mod = 1
```

Comment:
this typepath shouldnt exist

Reply:
why

Reply:
because the slot var exists

Reply:
yeah but pixel shift and other slots exist but i have a base one so i dont have to put slot and the same stuff as size_mod in example on all of them

Reply:
it should be done on the individual attachments